src/MoBi.Presentation/Tasks/OutputSelectionsTask.cs

```diff
@@ -0,0 +1,150 @@
+   }
+
+   public class OutputSelectionsTask : IOutputSelectionsTask
+   {
+      private readonly IOSPSuiteExecutionContext _context;
+      private readonly IMoBiApplicationController _applicationController;
+      private readonly ISimulationRepository _simulationRepository;
+      private readonly IDialogCreator _dialogCreator;
+
+      public OutputSelectionsTask(IOSPSuiteExecutionContext context, IMoBiApplicationController applicationController, ISimulationRepository simulationRepository, IDialogCreator dialogCreator)
+      {
+         _context = context;
+         _applicationController = applicationController;
+         _simulationRepository = simulationRepository;
+         _dialogCreator = dialogCreator;
+      }
+
+      public void AddOutputSelection(SimulationSettings simulationSettings, QuantitySelection preSelectedQuantitySelection = null)
+      {
+         var pathFromSimulations = selectPathFromSimulationsFor(AppConstants.Captions.AddDefaultCurveForNewSimulations, preSelectedQuantitySelection);
+         if(pathFromSimulations == null)
+            return;
+
+         if (!validateAndWarnNewPath(pathFromSimulations, simulationSettings))
+            return;
+
+         simulationSettings.OutputSelections.AddOutput(new QuantitySelection(pathFromSimulations, QuantityType.Undefined));
+         projectHasChanged();
+      }
+
+      private void showWarningMessage(ObjectPath pathFromSimulations)
+      {
+         _dialogCreator.MessageBoxInfo(AppConstants.Validation.ThePathIsAlreadySelectedAsAnOutput(pathFromSimulations));
+      }
+
+      private bool alreadyContainsPath(ObjectPath pathFromSimulations, SimulationSettings simulationSettings)
+      {
+         return simulationSettings.OutputSelections.Any(selection => Equals(selection.Path, pathFromSimulations.ToPathString()));
+      }
+
+      private void projectHasChanged()
+      {
+         _context.Project.HasChanged = true;
+      }
+
+      public void EditOutputSelection(SimulationSettings simulationSettings, QuantitySelection selection)
+      {
+         var newPath = selectPathFromSimulationsFor(AppConstants.Captions.ChangeDefaultCurveForNewSimulations, selection);
+
+         if (noChangesRequired(selection, newPath))
+            return;
+
+         if(validateAndUpdate(selection, newPath, simulationSettings))
+            projectHasChanged();
+      }
+
+      private static bool noChangesRequired(QuantitySelection selection, ObjectPath newPath)
+      {
+         return newPath == null || Equals(selection.Path, newPath.ToPathString());
+      }
+
+      private ObjectPath selectPathFromSimulationsFor(string caption, QuantitySelection preSelectedQuantity = null)
+      {
+         using (var modalPresenter = _applicationController.Start<IModalPresenter>())
+         {
+            modalPresenter.Text = caption;
+            var selectPathPresenter = _applicationController.Start<IHierarchicalQuantitySelectionPresenter>();
```

Comment:
All this Hierarchical Selection Presenter stuff is MoBi-specific. At first, I tried to move it to Core, but there were a few MoBi-specific things I couldn't move easily.

When I looked at the implementation of other SimulationSettingsItems presenters, they were all implemented in MoBi, so I didn't think it would be the worst idea to move output selections there as well.

I could have only added the interface ```IOutputSelectionsTask``` to Core, and implemented it in MoBi, but I chose to move the presenter to MoBi instead.